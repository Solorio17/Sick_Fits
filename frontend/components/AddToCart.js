import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Query, Mutation } from 'react-apollo';
import { CURRENT_USER_QUERY } from './User';

const ADD_TO_CART_MUTATION = gql`
    mutation addToCart($id: ID!){
        addToCart(id: $id){
            id
            quantity
        }
    }
`;

class AddToCart extends Component{
    render(){
        const { id } = this.props
        return(
            <Mutation 
                mutation={ADD_TO_CART_MUTATION} 
                variables={{
                    id: id
                }}
                refetchQueries={[{ query: CURRENT_USER_QUERY }]}
            >
                {(addToCart, { error, loading}) => <button className="adding" onClick={addToCart} disabled={loading}>Add{loading && 'ing'} to cart</button>}
            </Mutation>
        )
    }
}

export default AddToCart;
export { ADD_TO_CART_MUTATION };