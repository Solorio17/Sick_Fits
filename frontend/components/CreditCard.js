import React, { Component } from 'react';
import gql from 'graphql-tag';
import StripeCheckout from 'react-stripe-checkout';
import { Mutation } from 'react-apollo';
import Router from 'next/router';
import NProgress from 'nprogress';
import propTypes from 'prop-types';
import calcTotalPrice from '../lib/calcTotalPrice';
import Error from './ErrorMessage';
import User, { CURRENT_USER_QUERY } from './User';

const CREATE_ORDER_MUTATION = gql`
    mutation createOrder($token: String!){
        createOrder(token: $token){
            id
            charge
            total
            items{
                id
                title
            }
        }
    }
`;

function totalItems(cart){
    return cart.reduce((tally, cartItem) => tally + cartItem.quantity, 0)
}

class CreditCard extends Component{
    onToken = async (res, createOrder) =>{
        NProgress.start();
        // console.log(res.id)
        //manually call mutation once we have stripe token
        const order = await createOrder({
            variables: {
                token: res.id
            },
        }).catch(err => {
            alert(err.message)
        });
        Router.push({
            pathname: "/order",
            query: { id: order.data.createOrder.id }
        })
    }
    render(){
        return(
            <User>
                {({data: { me }, loading }) => {

                if(loading) return null;
                
                return (
                    <Mutation mutation={CREATE_ORDER_MUTATION} refetchQueries={[{ query: CURRENT_USER_QUERY }]}> 
                        {(createOrder) => (
                        
                        <StripeCheckout
                            amount={calcTotalPrice(me.cart)}
                            name="Sick Fits"
                            description={`Total items: ${totalItems(me.cart)}`}
                            image={me.cart.length && me.cart[0].item && me.cart[0].item.image}
                            stripeKey="pk_test_LPv396GsOHWwT37lbslxNgW3"
                            currency="USD"
                            email={me.email}
                            token={res => this.onToken(res, createOrder)}
                        >
                            {this.props.children}
                        </StripeCheckout>
                        )}
                    </Mutation>
                )}}
            </User>
        )
    }
}

export default CreditCard;
export { CREATE_ORDER_MUTATION };