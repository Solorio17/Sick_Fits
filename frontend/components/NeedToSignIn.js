import { Query } from 'react-apollo';
import { CURRENT_USER_QUERY } from './User';
import Signin from './Signin';

const NeedToSignIn = props =>{
    return(
        <Query query={CURRENT_USER_QUERY}>
            {({data, loading}) => {
                if(loading) return <p>Loading...</p>
                if(!data.me){
                    return(
                        <div>
                            <h3>Please sign in first!</h3>
                            <Signin/>
                        </div>
                    )
                }
                return props.children
            }}
        </Query>
    )
}

export default NeedToSignIn;