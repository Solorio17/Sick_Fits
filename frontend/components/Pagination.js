import React from 'react';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import { perPage } from '../config';
import Head from 'next/head'
import Link from 'next/link';
import PaginationStyles from '../components/styles/PaginationStyles';


const PAGINATION_QUERY = gql`
    query PAGINATION_QUERY {
        itemsConnection{
            aggregate{
                count
            }
        }
    }
`;


const Pagination = props =>{
    return(
            <Query query={PAGINATION_QUERY}>
                {({ data, loading, error}) =>{
                    if(loading) return <p>Loading...</p>
                    const count = data.itemsConnection.aggregate.count;
                    const pages = Math.ceil(count / perPage);
                    const page = props.page;
                    const previous = "<< Prev";
                    const next = "Next >>";

                    return(
                        <PaginationStyles>
                            <Head>
                                <title>Sick Fits - Page {props.page} of {pages}</title>
                            </Head>
                            <Link 
                                prefetch
                                href={{
                                pathname: 'items',
                                query: {page: page - 1}
                            }}>
                            <a className="prev" aria-disabled={page <= 1}>{previous}</a>
                            </Link>
                            <p>Page {page} of {pages}</p>
                            <p>{count} Items total</p>
                            <Link 
                                prefetch
                                href={{
                                pathname: 'items',
                                query: {page: page + 1}
                            }}>
                            <a className="next" aria-disabled={page >= pages}>{next}</a>
                            </Link>
                        </PaginationStyles>
                    )
                }}
            </Query>
    )
}

export default Pagination;