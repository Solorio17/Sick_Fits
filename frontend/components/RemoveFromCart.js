import React, { Component } from 'react';
import styled from 'styled-components';
import propTypes from 'prop-types';
import gql from 'graphql-tag';
import { CURRENT_USER_QUERY } from './User';
import { Mutation } from 'react-apollo';

const REMOVE_FROM_CART_MUTATION = gql`
    mutation removeFromCart($id: ID!){
        removeFromCart(id: $id){
            id
        }
    }
`;

const DeleteItemButton = styled.button`
    font-size: 18px;
    background: none;
    border: 0;
    &:hover{
        color: ${props => props.theme.red};
        cursor: pointer
    }
`;

class RemoveFromCart extends Component{
    static propTypes = { id: propTypes.string.isRequired };

    update = ( cache, payload ) => {
        //read cache
        const data = cache.readQuery({
            query: CURRENT_USER_QUERY
        })
        //remove the item from cart
        const cartItemId = payload.data.removeFromCart.id;
        data.me.cart = data.me.cart.filter(cartItem => cartItem.id !== cartItemId);
        //write it back to the cache
        cache.writeQuery({ query: CURRENT_USER_QUERY, data: data})
    }

    render(){
        return(
            <Mutation 
                mutation={REMOVE_FROM_CART_MUTATION} 
                variables={{id: this.props.id}} 
                update={this.update}
                optimisticResponse={{
                    __typename: 'Mutation',
                    removeFromCart: {
                        __typename: 'CartItem',
                        id: this.props.id
                    }
                }}
            >
                {(removeFromCart, { loading, error }) => (
                    <DeleteItemButton 
                        title="deleteItem" 
                        onClick={() => {removeFromCart().catch(error => alert(error.message))}} 
                        disabled={loading}
                    > &times; </DeleteItemButton>
                )}
            </Mutation>
        )
    }
}

export default RemoveFromCart;