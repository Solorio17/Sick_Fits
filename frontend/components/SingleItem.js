import React, { Component } from 'react';
import gql from 'graphql-tag';
import { Query } from 'react-apollo';
import Error from '../components/ErrorMessage';
import styled from 'styled-components';
import Head from 'next/head';

const SingleItemStyle = styled.div`
    max-width: 1200px;
    margin: 2rem auto;
    box-shadow: ${props => props.theme.bs};
    display: grid;
    grid-auto-columns: 1fr;
    grid-auto-flow: column;
    min-height: 800px;
    img{
        width: 100%;
        height: 100%;
        object-fit: contain;
    }
    .details{
        text-align: center;
    }
    .details h2{
        font-style: italic;
    }
    .details p span{
        font-weight: bold;
    }
`;

const SINGLE_ITEM_QUERY = gql`
    query SINGLE_ITEM_QUERY($id: ID!){
        item(where: { id: $id }){
            id
            title
            description
            price
            largeImage
        }
    }
`;

class SingleItem extends Component{
    render(){
        return(
            <Query query={SINGLE_ITEM_QUERY} variables={{
                id: this.props.id
            }}>
            {({error, loading, data}) => {
                if(error) return <Error error={error}/>
                if(loading) return <p>Loading...</p>
                if(!data.item) return <p>No item found for {this.props.id}</p>
                const item = data.item;
                return(
                    <SingleItemStyle>
                        <Head>
                            <title>Sick Fits | {item.title}</title>
                        </Head>
                        <img src={item.largeImage} alt={item.title}/>
                        <div className="details">
                            <h2>{item.title}</h2>
                            <p><span>More about this item: </span>{item.description}</p>
                            <h3>{"$" + item.price /100}</h3>
                        </div>
                    </SingleItemStyle>
                )
            }}
            </Query>
        )
    }
}

export default SingleItem;
export { SINGLE_ITEM_QUERY };