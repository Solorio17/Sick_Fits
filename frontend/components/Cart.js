import React from 'react';
import gql from 'graphql-tag';
import { Query, Mutation } from 'react-apollo';
import calcTotalPrice from '../lib/calcTotalPrice';
import formatMoney from '../lib/formatMoney';
import { adopt } from 'react-adopt';
import CartStyles from './styles/CartStyles';
import SickButton from './styles/SickButton';
import Supreme from './styles/Supreme';
import CloseButton from './styles/CloseButton';
import CartItem from './CartItem';
import User from './User';
import CreditCard from './CreditCard';

const LOCAL_STATE_QUERY = gql`
    query{
        cartOpen @client
    }
`;

const TOGGLE_CART_MUTATION = gql`
    mutation{
        toggleCart @client
    }
`;

const Composed = adopt({
    user: ({ render }) => <User>{render}</User>,
    toggleCart: ({ render }) => <Mutation mutation={TOGGLE_CART_MUTATION}>{render}</Mutation>,
    localState: ({ render }) => <Query query={LOCAL_STATE_QUERY}>{render}</Query>
})

const Cart = () => {
    return(
        <Composed>
            {({user, toggleCart, localState}) => {
                const me = user.data.me;
                if(!me) return null
                return(
                    <CartStyles open={localState.data.cartOpen}>
                        <header>
                            <CloseButton onClick={toggleCart} title="close">X</CloseButton>
                            <Supreme>{me.name}'s Cart</Supreme>
                            <p>You have {me.cart.length} different item{me.cart.length === 1 ? '' : 's'} in your cart</p>
                        </header>
                        
                        <ul>{me.cart.map(cartItem => <CartItem key={cartItem.id} cartItem={cartItem}/>)}</ul>

                        <footer>
                            <p>Total: {formatMoney(calcTotalPrice(me.cart))}</p>
                            <CreditCard>
                                <SickButton>Checkout</SickButton>
                            </CreditCard>
                        </footer>
                    </CartStyles>
                )
            }}
        </Composed>
    )
}

export default Cart;
export { LOCAL_STATE_QUERY, TOGGLE_CART_MUTATION };