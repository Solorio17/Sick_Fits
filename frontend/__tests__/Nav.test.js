import { CURRENT_USER_QUERY } from '../components/User';
import NeedToSignIn from '../components/NeedToSignIn';
import { shallow, mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeUser, fakeCartItem } from '../lib/testUtils';
import Nav from '../components/Nav';

const notSignedInMocks = [
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: {me : null}}
    }
]

const SignedInMocks = [
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: {me : fakeUser()}}
    }
]

const SignedInMocksWithCartItems = [
    {
        request: { query: CURRENT_USER_QUERY },
        result: { data: {me : {
            ...fakeUser(),
            cart: [fakeCartItem(), fakeCartItem()]
        }}}
    }
]

describe('Nav component', ()=>{
    it('renders the minimal nav links when logged out', async()=>{
        const wrapper = mount(
            <MockedProvider mocks={notSignedInMocks}>
                <Nav/>
            </MockedProvider>
        );
        // console.log(wrapper.debug())
        await wait();
        wrapper.update();
        // console.log(wrapper.debug())
        const nav = wrapper.find('ul[data-test="nav"]');
        // console.log(nav.debug())
        expect(nav.children().length).toBe(2)
        expect(nav.text()).toContain('Sign In')
        // expect(toJSON(nav)).toMatchSnapshot();
    });

    it('renders the rest of the nav links when logged in', async()=>{
        const wrapper = mount(
            <MockedProvider mocks={SignedInMocks}>
                <Nav/>
            </MockedProvider>
        );
        // console.log(wrapper.debug())
        await wait();
        wrapper.update();
        // console.log(wrapper.debug())
        const nav = wrapper.find('ul[data-test="nav"]');
        // console.log(nav.debug())
        expect(toJSON(nav)).toMatchSnapshot();
    });

    it('renders the rest of the nav links when logged in with cart items', async()=>{
        const wrapper = mount(
            <MockedProvider mocks={SignedInMocksWithCartItems}>
                <Nav/>
            </MockedProvider>
        );
        // console.log(wrapper.debug())
        await wait();
        wrapper.update();
        // console.log(wrapper.debug())
        const nav = wrapper.find('[data-test="nav"]');
        // console.log(nav.debug())
        const count = nav.find('div.count');
        expect(toJSON(count)).toMatchSnapshot();

    });
})