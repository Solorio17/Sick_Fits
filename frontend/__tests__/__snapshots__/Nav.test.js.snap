// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders the rest of the nav links when logged in 1`] = `
<ul
  className="NavStyles-sc-11c0d2g-0 gKhvNz"
  data-test="nav"
>
  <Link
    href="/items"
  >
    <a
      href="/items"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
  <Link
    href="/sell"
  >
    <a
      href="/sell"
      onClick={[Function]}
    >
      Sell
    </a>
  </Link>
  <Link
    href="/orders"
  >
    <a
      href="/orders"
      onClick={[Function]}
    >
      Orders
    </a>
  </Link>
  <Link
    href="/account"
  >
    <a
      href="/account"
      onClick={[Function]}
    >
      Account
    </a>
  </Link>
  <Signout>
    <Mutation
      mutation={
        Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "name": Object {
                "kind": "Name",
                "value": "SIGNOUT_MUTATION",
              },
              "operation": "mutation",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "alias": undefined,
                    "arguments": Array [],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "signout",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "message",
                          },
                          "selectionSet": undefined,
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 85,
            "start": 0,
          },
        }
      }
      refetchQueries={
        Array [
          Object {
            "query": Object {
              "definitions": Array [
                Object {
                  "directives": Array [],
                  "kind": "OperationDefinition",
                  "name": undefined,
                  "operation": "query",
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "me",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "name",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "email",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "permissions",
                              },
                              "selectionSet": undefined,
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "orders",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "cart",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "id",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "quantity",
                                    },
                                    "selectionSet": undefined,
                                  },
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "item",
                                    },
                                    "selectionSet": Object {
                                      "kind": "SelectionSet",
                                      "selections": Array [
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "id",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "price",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "image",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "title",
                                          },
                                          "selectionSet": undefined,
                                        },
                                        Object {
                                          "alias": undefined,
                                          "arguments": Array [],
                                          "directives": Array [],
                                          "kind": "Field",
                                          "name": Object {
                                            "kind": "Name",
                                            "value": "description",
                                          },
                                          "selectionSet": undefined,
                                        },
                                      ],
                                    },
                                  },
                                ],
                              },
                            },
                          ],
                        },
                      },
                    ],
                  },
                  "variableDefinitions": Array [],
                },
              ],
              "kind": "Document",
              "loc": Object {
                "end": 417,
                "start": 0,
              },
            },
          },
        ]
      }
    >
      <button
        onClick={[Function]}
      >
        Sign Out
      </button>
    </Mutation>
  </Signout>
  <Mutation
    mutation={
      Object {
        "definitions": Array [
          Object {
            "directives": Array [],
            "kind": "OperationDefinition",
            "name": undefined,
            "operation": "mutation",
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [
                    Object {
                      "arguments": Array [],
                      "kind": "Directive",
                      "name": Object {
                        "kind": "Name",
                        "value": "client",
                      },
                    },
                  ],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "toggleCart",
                  },
                  "selectionSet": undefined,
                },
              ],
            },
            "variableDefinitions": Array [],
          },
        ],
        "kind": "Document",
        "loc": Object {
          "end": 48,
          "start": 0,
        },
      }
    }
  >
    <button
      onClick={[Function]}
    >
      My Cart
      <CartCount
        count={0}
      >
        <CartCount__AnimationStyles>
          <span
            className="CartCount__AnimationStyles-xxvp4g-0 dJLOrG"
          >
            <TransitionGroup
              childFactory={[Function]}
              component="div"
            >
              <div>
                <CSSTransition
                  className="count"
                  classNames="count"
                  in={true}
                  key=".$0"
                  onExited={[Function]}
                  timeout={
                    Object {
                      "enter": 500,
                      "exit": 500,
                    }
                  }
                  unmountOnExit={true}
                >
                  <Transition
                    appear={false}
                    className="count"
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={
                      Object {
                        "enter": 500,
                        "exit": 500,
                      }
                    }
                    unmountOnExit={true}
                  >
                    <CartCount__CountDot
                      className="count"
                    >
                      <div
                        className="count CartCount__CountDot-xxvp4g-1 kNjNtw"
                      >
                        0
                      </div>
                    </CartCount__CountDot>
                  </Transition>
                </CSSTransition>
              </div>
            </TransitionGroup>
          </span>
        </CartCount__AnimationStyles>
      </CartCount>
    </button>
  </Mutation>
</ul>
`;

exports[`Nav component renders the rest of the nav links when logged in with cart items 1`] = `
<div
  className="count CartCount__CountDot-xxvp4g-1 kNjNtw"
>
  6
</div>
`;
